ee/desktop/runner/server/server.go

```diff
@@ -0,0 +1,148 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/kolide/kit/ulid"
+)
+
+// RunnerServer provides IPC for user desktop processes to communicate back to the root desktop runner.
+// It allows the user process to notify and monitor the health of the runner process.
+type RunnerServer struct {
+	server                          *http.Server
+	listener                        net.Listener
+	logger                          log.Logger
+	desktopProcAuthTokens           map[string]string
+	mutex                           sync.Mutex
+	controlRequestIntervalOverrider controlRequestIntervalOverrider
+}
+
+const (
+	HealthCheckEndpoint                = "/health"
+	MenuOpenedEndpoint                 = "/menuopened"
+	controlRequestAccelerationInterval = 5 * time.Second
+	controlRequestAcclerationDuration  = 1 * time.Minute
+)
+
+type controlRequestIntervalOverrider interface {
+	SetControlRequestIntervalOverride(time.Duration, time.Duration)
+}
+
+func New(logger log.Logger, controlRequestIntervalOverrider controlRequestIntervalOverrider) (*RunnerServer, error) {
+	listener, err := net.Listen("tcp", "localhost:0")
```

Comment:
Does using port zero mean it will choose a dynamically generated unused port?

Reply:
yea, it just picks a random port, it's what the current "monitor server" that is doing now on stable